scheduler/src/cook/mesos/api.clj

```diff
@@ -148,22 +148,20 @@
    (s/optional-key :extract) s/Bool
    (s/optional-key :cache) s/Bool})
 
-(def Instance
-  "Schema for a description of a single job instance."
-  {:status s/Str
-   :task_id s/Uuid
-   :executor_id s/Uuid
-   :slave_id s/Str
-   :hostname s/Str
-   :preempted s/Bool
-   :backfilled s/Bool
-   :ports [s/Int]
-   (s/optional-key :start_time) s/Int
-   (s/optional-key :end_time) s/Int
-   (s/optional-key :reason_code) s/Int
-   (s/optional-key :output_url) s/Str
-   (s/optional-key :cancelled) s/Bool
-   (s/optional-key :reason_string) s/Str})
+(def Command
+  "Schema for a command"
+  {:value s/Str
+   (s/optional-key :async?) s/Bool
+   (s/optional-key :guard?) s/Bool
+   (s/optional-key :default?) s/Bool})
+
+(def CommandRequest
+  "Schema for a command in request"
+  {:value s/Str
+   (s/optional-key :name) s/Str
```

Comment:
The name is generated on submit, combining the command order with the command suffix (before or after). It's used to create log files per command. The executor sends it back when updating task status, so it's included in the response from the scheduler API.

Here's where it's generated:
https://github.com/mojotech/Cook/blob/7c0c8162a136f032d37fe01b8686ee0bb7b9edce/scheduler/src/cook/mesos/util.clj#L168